File: evidence/generated-systems/golang/manufacturing-on-golang/manufacturing-on-golang/internal/model/Plant.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Plant Declaration
//==============================================================
type Plant struct {
    gorm.Model
     Name                                    string
    PlantCode                                    string
    Address                                                            string
    TimeZone                                    string
    EnterpriseId         *uint
    Enterprise           *Enterprise `gorm:"foreignKey:EnterpriseId"`
     ProductionLines           []ProductionLine `gorm:"foreignKey:ProductionLinesFromPlantId"`
     WorkCenters           []WorkCenter `gorm:"foreignKey:WorkCentersFromPlantId"`
     Warehouses           []Warehouse `gorm:"foreignKey:WarehousesFromPlantId"`
     Assets           []Asset `gorm:"foreignKey:AssetsFromPlantId"`
     ProductionSchedules           []ProductionSchedule `gorm:"foreignKey:ProductionSchedulesFromPlantId"`

// parent associations as their child

}

File: evidence/generated-systems/angular/manufacturing-on-angular/src/app/models/Plant.js

// Define collection and schema for Plant
export interface Plant {
    name:
	type : string
    plantCode:
	type : string
    address:
	type : Address
    timeZone:
	type : string
    Enterprise:
	type : Schema.Types.ObjectId
    ProductionLines:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ProductionLine' }]
    WorkCenters:
 	type : [{ type: Schema.Types.ObjectId, ref: 'WorkCenter' }]
    Warehouses:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Warehouse' }]
    Assets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Asset' }]
    ProductionSchedules:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ProductionSchedule' }]
#
    collection: 'plants'
}
